### PR TITLE
Coadd_extract

### PR DESCRIPTION
**Pull request by SunilSimha**
A simple amendment to automatically check if optimal extraction was not present and hence use boxcar while warning the user of the action.

### CODE REVIEW

File: pypit/scripts/coadd_1dspec.py [hunk 1]
```diff
@@ -95,6 +95,12 @@ def main(args, unit_test=False, path=''):
             if mtch_obj is None:
                 print("No object {:s} in file {:s}".format(iobj, fkey))
             elif len(mtch_obj) == 1:
+                #Check if optimal extraction is present in all  objects. If not, warn the user and set ex_value to 'box'.
```
Comment (profxj) [inline pypit/scripts/coadd_1dspec.py]: Is it whether the object is present or whether it is full of
NAN?  Or, is NAN what we get when optimal extraction
is not performed?

File: pypit/scripts/coadd_1dspec.py [hunk 2]
```diff
@@ -95,6 +95,12 @@ def main(args, unit_test=False, path=''):
...
+                hdulist = fits.open(fkey)
+                obj_opt_flam = hdulist[mtch_obj[0]].data['OPT_FLAM']
+                if any(isnan(obj_opt_flam)):
```
Comment (profxj) [inline pypit/scripts/coadd_1dspec.py]: Can you use np.any here?  I'm not familiar with
Python's any.

**Comment by profxj** [comment]
Thanks for this @SunilSimha .  Have put in 2 minor suggestions
for edits.

Addresses Issue #242

**Comment by SunilSimha** [comment]
@profxj, np.isnan returns a boolean array. It is true if the corresponding
array element is NaN. any() returns true if at least one element of an
array is true. So the code checks if there is at least one NaN in the opt
extraction and changes the method to boxcar if true. Is np.any
different/faster?

Sunil

**Comment by profxj** [comment]
Ok, Python's *any* sounds fine.

But how about my main query is whether 
the optimal extraction should be missing 
altogether instead of being NAN ?

**Comment by SunilSimha** [comment]
I don't really know how the 1dspec files are generated. As I understand,
the files will have a NaN array in OPT_FLAM in case the optimal extract was
not produced. I haven't come across a case where the OPT_FLAM array doesn't
exist at all. Is this possible? If so, I can modify the code suitably.

Sunil

**Comment by profxj** [comment]
Looking at the code in arsave.py, it seems possible to not have
either array (box or optimal).  So, since we are here, please add
in a 'catch' for cases without one or the other.    Thanks!

**Comment by SunilSimha** [comment]
@profxj, I've added a try/except statement to catch that error. Is this alright?

**Comment by profxj** [comment]
Looks great.  Am merging

**Pull request merged**